Plugins/Sycl/src/Seeding/TripletSearch.hpp

```diff
@@ -204,7 +204,10 @@ class TripletSearch {
           const auto top = deviceIndTopDuplets[it];
           // this will be the t-th top space point for
           // fixed middle and bottom SP
-          auto t = m_countTripletsAcc[ib].fetch_add(1);
+          vecmem::device_vector<uint32_t> deviceCountTriplets(
+              m_countTripletsView);
+          vecmem::atomic obj(&deviceCountTriplets[ib]);
```

Comment:
Wow... This actually works? :smile: That's great!

As you know, [vecmem::atomic](https://github.com/acts-project/vecmem/blob/main/core/include/vecmem/memory/atomic.hpp) is a template. I know that C\+\+17 (and oneAPI in particular) can do this. (https://en.cppreference.com/w/cpp/language/class_template_argument_deduction) I just never tried...

I was pondering for a moment whether to allow this to go through, or to ask you to change `vecmem::atomic` to `vecmem::atomic<uint32_t>`. But you know what, let's keep it as-is. Let's not worry about SYCL implementations that do not support CTAD...